async/mailbox.go

```diff
@@ -115,6 +122,7 @@ func (bx *Mailbox) ProcessMessages() {
 			unCompletedMsgs = append(unCompletedMsgs, msg)
 		}
 	}
+	log.Infof("processed %d (worker) messages in %v", len(bx.msgs), time.Since(start))
```

Comment:
can we pass the stats handler through to here too & emit a latency gauge for each message callback?